routing/turns_generator.cpp

```diff
@@ -299,45 +299,51 @@ TurnDirection FindDirectionByAngle(vector<pair<double, TurnDirection>> const & l
- * \param segment is a ingoing or outgoing feature segment.
- * \param ft is a ingoing or outgoing feature.
+ * \param path geometry of the segment.
  * \param junctionPoint is a junction point.
  * \param maxPointsCount returned poit could't be more than maxPointsCount poins away from junctionPoint
  * \param minDistMeters returned point should be minDistMeters away from junctionPoint if ft is long and consists of short segments
  * \param GetPointIndex is a function for getting points by index.
- * It defines a direction of following along a feature. So it differs for ingoing and outgoing cases.
+ * It defines a direction of following along a feature. So it differs for ingoing and outgoing
+ * cases.
  * It has following parameters:
- * - start is an index of the start point of a feature segment. For example, FtSeg::m_pointStart.
- * - end is an index of the end point of a feature segment. For example, FtSeg::m_pointEnd.
+ * - start is an index of the start point of a feature segment. For example, path.back().
+ * - end is an index of the end point of a feature segment. For example, path.front().
  * - shift is a number of points which shall be added to end or start index. After that
  *   the sum reflects an index of a feature segment point which will be used for a turn calculation.
  * The sum shall belongs to a range [min(start, end), max(start, end)].
  * shift belongs to a  range [0, abs(end - start)].
  * \return an ingoing or outgoing point for a turn calculation.
  */
-m2::PointD GetPointForTurn(OsrmMappingTypes::FtSeg const & segment, FeatureType const & ft,
-                           m2::PointD const & junctionPoint,
-                           size_t const maxPointsCount,
-                           double const minDistMeters,
-                           size_t (*GetPointIndex)(const size_t start, const size_t end, const size_t shift))
+m2::PointD GetPointForTurn(vector<m2::PointD> const & path, m2::PointD const & junctionPoint,
+                           size_t const maxPointsCount, double const minDistMeters,
+                           size_t (*GetPointIndex)(const size_t start, const size_t end,
+                                                   const size_t shift))
 {
   double curDistanceMeters = 0.;
   m2::PointD point = junctionPoint;
   m2::PointD nextPoint;
 
-  size_t const numSegPoints = abs(segment.m_pointEnd - segment.m_pointStart);
+  size_t const numSegPoints = path.size() - 1;
   ASSERT_GREATER(numSegPoints, 0, ());
-  ASSERT_LESS(numSegPoints, ft.GetPointsCount(), ());
   size_t const usedFtPntNum = min(maxPointsCount, numSegPoints);
 
   for (size_t i = 1; i <= usedFtPntNum; ++i)
   {
-    nextPoint = ft.GetPoint(GetPointIndex(segment.m_pointStart, segment.m_pointEnd, i));
+    nextPoint = path[GetPointIndex(0, numSegPoints, i)];
+
+    // TODO The code below is a stub for compatability with older versions with this function.
```

Comment:
Could you please clarify what compatibility means in this context? New version of the App will have new function and will not have older versions of the function.